src/main/java/org/takes/rq/multipart/RqMtBase.java

```diff
@@ -125,27 +127,24 @@
      * Ctor.
      * @param req Original request
      * @throws IOException If fails
-     * @todo #950:30m Remove code from this ctor, leaving only
-     *  initialization. Currently this constructor access body
-     *  of the request and triggers its evaluation. This breaks
-     *  composition of multiple request, as it can be seen in
-     *  {@link RqMtFake}. When this task is done, remove
-     *  explicit lazy evaluation for RqMtFake.
      * @checkstyle ExecutableStatementCountCheck (2 lines)
      */
     public RqMtBase(final Request req) throws IOException {
         this.origin = req;
-        this.stream = new RqLengthAware(req).body();
-        this.buffer = ByteBuffer.allocate(
-            // @checkstyle MagicNumberCheck (1 line)
-            Math.min(8192, this.stream.available())
-        );
-        this.map = this.requests(req);
+        this.sstream = new Sticky<>(() -> new RqLengthAware(req).body());
```

Comment:
@baudoliver7 Is it possible to use `RqOnce` on the ctor parameter instead of three Stickies? 
It seems also that some fields could be moved inside of `request` method, such as `sbuffer`